packages/attune/src/server/repo/index/sign.rs

```diff
@@ -706,120 +756,172 @@ pub async fn handler(
         }
     }
 
-    // Upload the updated Packages index file.
+    // Upload the updated Packages index file to standard path and all by-hash paths concurrently
+    let by_hash_prefix = format!(
+        "{}/dists/{}/{}/binary-{}/by-hash",
+        repo.s3_prefix,
+        req.change.distribution,
+        result.changed_packages_index.component,
+        result.changed_packages_index.architecture
+    );
     if result.changed_packages_index_contents.is_empty() {
-        state
-            .s3
-            .delete_object()
-            .bucket(&repo.s3_bucket)
-            .key(format!(
+        let deletions = [
+            format!(
                 "{}/dists/{}/{}/binary-{}/Packages",
                 repo.s3_prefix,
                 req.change.distribution,
                 result.changed_packages_index.component,
                 result.changed_packages_index.architecture
-            ))
-            .send()
-            .await
-            .unwrap();
+            ),
+            format!(
+                "{}/SHA256/{}",
+                by_hash_prefix, result.changed_packages_index.sha256sum
+            ),
+            format!(
+                "{}/SHA1/{}",
+                by_hash_prefix, result.changed_packages_index.sha1sum
+            ),
+            format!(
+                "{}/MD5Sum/{}",
+                by_hash_prefix, result.changed_packages_index.md5sum
+            ),
+        ]
+        .into_iter()
+        .map(|key| {
+            state
+                .s3
+                .delete_object()
+                .bucket(&repo.s3_bucket)
+                .key(key)
+                .send()
+        });
+        for deletion in futures_util::future::join_all(deletions).await {
+            if let Err(err) = deletion {
+                tracing::warn!("Failed to delete object during index cleanup: {err:?}");
+            }
+        }
```

Comment:
These deletions still have to occur after the Release file is updated. Otherwise, the current Release file will reference an index that no longer exists.

Reply:
Ah, yep, I was thrown off by the existing code deleting the `Packages` file here. but I think that wasn't valid either- I've moved into a two-phase style: https://github.com/attunehq/attune/pull/82/commits/065da35fe05ba2a31c3091a769f7e32839f27093, specifically this comment: https://github.com/attunehq/attune/blob/ef83e6911199b63155750630e2d8c3bb7d4061dc/packages/attune/src/server/repo/index/sign.rs#L750-L755